src/jit/codegenarm64.cpp

```diff
@@ -5294,22 +5236,25 @@ void CodeGen::genCallInstruction(GenTreePtr node)
         // Load the call target address in x16
         instGen_Set_Reg_To_Imm(EA_8BYTE, REG_IP0, (ssize_t) addr);
 
-        // indirect call to constant address in IP0
-        genEmitCall(emitter::EC_INDIR_R,
-                    methHnd, 
-                    INDEBUG_LDISASM_COMMA(sigInfo)
-                    nullptr, //addr
-                    retSize,
-                    ilOffset,
-                    REG_IP0);
+        // indirect call to constant address in IP0 (using the direct call dispatch).
+        genEmitCall(GenEmitCallInfo(GenCallTypeToEmit::GEN_EMIT_CALL_DIRECT,
+                                    emitter::EC_INDIR_R,
+                                    methHnd,
+                                    INDEBUG_LDISASM_COMMA(sigInfo)
+                                    nullptr, // addr
+                                    callReturnTypes,
+                                    ilOffset,
+                                    REG_IP0));
 #else
         // Non-virtual direct call to known addresses
-        genEmitCall(emitter::EC_FUNC_TOKEN,
-                    methHnd, 
-                    INDEBUG_LDISASM_COMMA(sigInfo)
-                    addr,
-                    retSize,
-                    ilOffset);
+        genEmitCall(GenEmitCallInfo(GenCallTypeToEmit::GEN_EMIT_CALL_DIRECT,
```

Comment:
The two calls to genEmitCall(struct) can now be removed and one call can be placed after the if-then-else.